content/articles/testing-angular-app-with-protractor-and-jasmine/index.md

```diff
@@ -0,0 +1,243 @@
+
+
+```javascript
+describe("Testing exercise for protractor and Javascript application", function(){
+    it("Navigate to Blender Video search for videos",  async function(done){
+        let url = "https://video.blender.org/"
+        await browser.get(url);
+
+        expect(await browser.getCurrentUrl()).toBe(url);
+
+        let searchBar = element(by.css('#search-video'));
+        let searchButton = element(by.css('.icon-search'));
+        let searchResult = element(by.css('.search-result'));
+        
+         
+        searchBar.sendKeys("free");
+
+        searchButton.click();
+        
+        expect(searchResult.getCssValue('display')).toBe('block');
+
+        done();
+    })
+})
+```
+* We then make a further tests for a none existing search result ("neverdwelt"). We do not clear the text in the search bar so our search gets appended to the existing "free" text in the search bar. If you want to clear the search bar so as to make a new search uses the code.
+
+```javascript
+    searchBar.clear().then(function() {
+    searchBar.sendKeys('neverdwelt');
+    })
+
+```
+
+So, without changing the existing text, we append to it "neverdwelt" and programmatically trigger the search button. We are expecting the same outcome as the first, it should display the search result only in this case there are no elements in the list 
+
+
+```javascript
+describe("Testing exercise for protractor and Javascript application", function(){
+    it("Navigate to Blender Video search for videos",  async function(done){
+        let url = "https://video.blender.org/"
+        await browser.get(url);
+
+        expect(await browser.getCurrentUrl()).toBe(url);
+
+        let searchBar = element(by.css('#search-video'));
+        let searchButton = element(by.css('.icon-search'));
+        let searchResult = element(by.css('.search-result'));
+        
+         
+        searchBar.sendKeys("free");
+
+        searchButton.click();
+        
+        expect(searchResult.getCssValue('display')).toBe('block');
+        
+        searchBar.sendKeys('neverdwelt');
+        searchButton.click();
+        
+        expect(searchResult.getCssValue('display')).toBe('block');
+        done();
+    })
+})
+```
+
+You should see a Chrome browser window open up and navigate to the webpage specified perform a search and carry out the test. The test output should be 1 test, 1 assertion, 0 failures. Congratulations, you've run your first Protractor test!
+
+### Conclusion
+There are a lot of options present which we can use to capture DOM elements so we can use and manipulate them.
+Also available are methods that can be used to provide interaction with the elements on the page.
```

Comment:
Conclusion could have been a lot more descriptive.

Give a summary of what was covered, the key takeaways.